### PR TITLE
Add replay tool

### PR DESCRIPTION
**Pull request by ecourtois**
Adapt the simulator to new trace thread of mcu-firmware.
Optimize buoys by replacing them by cone entities.
Add the replay tool to replay a game saved in a trace file.

### CODE REVIEW

**Comment by ecourtois** [comment]
A trace file example to play with the new tool :
[TRACE.TXT](https://github.com/cogip/simulation/files/6580281/TRACE.TXT)

File: cogip/tools/replay/__main__.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+from PySide2 import QtWidgets
+
+from cogip import logger
+from cogip.entities.robot import RobotEntity, RobotShadowEntity
+from cogip.entities.table import TableEntity
+
+from .mainwindow import MainWindow
+
+
```
Comment (gdoffe) [inline cogip/tools/replay/__main__.py:15]: you can remove one empty line.

**Review by gdoffe — CHANGES_REQUESTED** [review]
https://github.com/cogip/simulation/pull/15/commits/2d51e13f0ddfd9f8122f735dea989cc9aaa92dde
"The input **is** parsed into"

**Review by gdoffe — APPROVED** [review]
LGTM ! nice job !

**Pull request merged**